app/furniture/marketplace/stripe_events_controller.rb

```diff
@@ -24,6 +24,19 @@ def create
         order.update!(status: :paid, placed_at: DateTime.now, payment_processor_fee_cents: balance_transaction.fee)
         order.events.create(description: "Payment Received")
 
+        square_client = Square::Client.new(
+          access_token: marketplace.square_access_token,
+          environment: "sandbox"
+        )
+
+        order_body = build_square_create_order_body(order)
+        square_client.orders.create_order(order_body)
+        payment_body = build_square_create_order_payment_body(order, marketplace, balance_transaction)
+
+        # DEV NOTE: Square requires that orders are paid in order to show up in the Seller Dashboard
+        square_client.payments.create_payment(payment_body)
+
+        order_to_square_order
```

Comment:
This controller is not a great example; because does way too much and all that stuff is Very Important.

Would you be available to extract this to a method on the `Order` class for now? (Or, alternatively, a `Marketplace::SquareOrder` or `Marketplace::Order::InSquare` class?)

Reply:
Definitely. I was prepared to go in that route after sketching out this call sequence. There's a few ways I can think of to create a better abstraction here. Allow me to enumerate out loud as a reference: 

1. Move logic to the existing `Marketplace::Order` model (as you suggest)
1. Move logic to a new model/class to encapsulate the `SquareOrder` behavior (as you suggest)
1. Similar to the above, move logic to a new single-purpose `SquareOrderCreateService` "service object" type class which executes this workflow and could be invoked like `SquareOrderCreateService.new(...).call` (inspired by some recent reading I've done at https://jardo.dev/rails-service-objects)
1. (More avant-garde for the moment): Rejiggle this controller to pass the event on to  an marketplace-level event bus (implemented as a _class_ with similar case/switch reducer) and inline the logic (or new class call) there.

At this nascent stage I like the idea of moving some business logic onto the existing `Marketplace::Order` class, and continue monitoring what types of behavior accumulate there. Rather than give into the seduction of decoupling prematurely. Then let's see how the resulting _specs as documentation_ turn out. Will it be easy to test? Will the tests feel like a satisfactory observation of the behavior in action with the model as the subject?